From 2b706ad05d3cde8639ce5f96e92976b7368b8abd Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 10:25:38 -0700
Subject: [PATCH 24/96] HBASE-7274 Enable JMX metrics collection for REST Server

Reason: Improvement
Author: Jimmy Xiang
Ref: CDH-8762
(cherry picked from commit 1c818adca5e06fe27b5ddef80ff6ceaa396feaeb)
---
 bin/hbase-config.sh |    9 +++++++++
 conf/hbase-env.sh   |    1 +
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/bin/hbase-config.sh b/bin/hbase-config.sh
index 149d515..c4a95ca 100644
--- a/bin/hbase-config.sh
+++ b/bin/hbase-config.sh
@@ -84,6 +84,15 @@ if [ -z "$HBASE_THRIFT_OPTS" ]; then
   export HBASE_THRIFT_OPTS="$HBASE_THRIFT_JMX_OPTS"
 fi
 
+# REST JMX opts
+if [ -z "$HBASE_REST_JMX_OPTS" ]; then
+  HBASE_REST_JMX_OPTS="$HBASE_JMX_OPTS -Dcom.sun.management.jmxremote.port=10105"
+fi
+# REST opts
+if [ -z "$HBASE_REST_OPTS" ]; then
+  export HBASE_REST_OPTS="$HBASE_REST_JMX_OPTS"
+fi
+
 # Source the hbase-env.sh.  Will have JAVA_HOME defined.
 if [ -f "${HBASE_CONF_DIR}/hbase-env.sh" ]; then
   . "${HBASE_CONF_DIR}/hbase-env.sh"
diff --git a/conf/hbase-env.sh b/conf/hbase-env.sh
index 7b5d0fd..86f388b 100644
--- a/conf/hbase-env.sh
+++ b/conf/hbase-env.sh
@@ -69,6 +69,7 @@ export HBASE_OPTS="-XX:+UseConcMarkSweepGC"
 # export HBASE_REGIONSERVER_OPTS="$HBASE_REGIONSERVER_OPTS $HBASE_JMX_BASE -Dcom.sun.management.jmxremote.port=10102"
 # export HBASE_THRIFT_OPTS="$HBASE_THRIFT_OPTS $HBASE_JMX_BASE -Dcom.sun.management.jmxremote.port=10103"
 # export HBASE_ZOOKEEPER_OPTS="$HBASE_ZOOKEEPER_OPTS $HBASE_JMX_BASE -Dcom.sun.management.jmxremote.port=10104"
+# export HBASE_REST_OPTS="$HBASE_REST_OPTS $HBASE_JMX_BASE -Dcom.sun.management.jmxremote.port=10105"
 
 # File naming hosts on which HRegionServers will run.  $HBASE_HOME/conf/regionservers by default.
 # export HBASE_REGIONSERVERS=${HBASE_HOME}/conf/regionservers
-- 
1.7.0.4

